packaging/Microsoft.NET.Sdk.WindowsDesktop/Microsoft.NET.Sdk.WindowsDesktop.ArchNeutral.csproj

```diff
@@ -20,9 +20,30 @@
     <PackageDescription>WindowsDesktop SDK</PackageDescription>
   </PropertyGroup>
 
+  <PropertyGroup>
+    <_WinFormsNuGetPath>$(PkgMicrosoft_Private_Winforms)</_WinFormsNuGetPath>
+    <_WinFormsContentPath>$(_WinFormsNuGetPath)\Wpf.Sdk.WindowsDesktop\*</_WinFormsContentPath>
+  </PropertyGroup>
+
   <ItemGroup>
     <PackagingContent Include="Sdk\*" SubFolder="root\Sdk" />
     <PackagingContent Include="targets\*" SubFolder="root\targets" />
     <PackagingContent Include="useSharedDesignerContext.txt" SubFolder="root" />
+
+    <!-- Windows Forms specific -->
+    <PackagingContent Include="$(_WinFormsContentPath)" SubFolder="root\targets" />
   </ItemGroup>
+
+  <Target Name="_EnsureWindowsFormsPackagingContent" BeforeTargets="IdentifyPackageAssets">
```

Comment:
This is an antipattern. Foo.*ArchNeutral.csproj, Foo.csproj etc. have certain conventions, and they weren't intended to be spots for additional domain-specific targets. Our domain-specific targets exist under a certain hierarchy under eng\ that I've already alluded to.

Reply:
I haven't read the docs you've shared, but my sense/guess here is that analyzer related props/targets is a new category of payload that's being introduced. If this is right... then consider introducing a new `Item` name, something like `AnalyzerPackagingContent` that's used to track these items. These would become agnostic of WPF/Winforms etc. far as build-logic goes.

Then in `Packaging.targets`, add logic specific to validation of these. For e.g., anything that's been added to `AnalyzerPackagingContent` - ensure that they exist, make sure they're added to `PackagingContent` etc.. Then all you'd do in `*.WindowsDesktop.ArchNeutral.csproj` is adding your props/targets to the newly introduced `AnalyzerPackagingContent` `Item` and let the backend targets do its work as before.

Reply:
My understanding that once https://github.com/dotnet/designs/pull/181 is approved and implemented none of this will be required. @ericstj am I reading your proposal correctly?

Reply:
I added this target purely to perform verifcations as you suggested earlier.
All work is done by these targets https://github.com/RussKie/winforms/blob/project_configuration/src/System.Windows.Forms.Analyzers/src/System.Windows.Forms.Analyzers.targets

Reply:
The reason I added this target in this project and not in Directory.Build.targets at other levels is these validations is only relevant to this project. I'm happy to move it out to a separate targets file under eng\ folder and reference it in this project.

Reply:
A different architecture is fine, but this project is not the right place for adding targets, imports etc. - if you need to extend or hook the build system, it has to be done elsewhere.

Reply:
Does 9c0757e7519cff48333b376866d417cd0e1da3b5 look better?